src/main/java/no/unit/nva/cristin/projects/NvaProjectBuilder.java

```diff
@@ -35,39 +35,69 @@
     private static final Map<String, String> cristinRolesToNva = Map.of("PRO_MANAGER", "ProjectManager",
         "PRO_PARTICIPANT", "ProjectParticipant");
 
+    private final transient CristinProject cristinProject;
+    private final transient NvaProject nvaProject;
+
+    public NvaProjectBuilder(CristinProject cristinProject) {
+        this.cristinProject = cristinProject;
+        nvaProject = new NvaProject();
+    }
+
     /**
      * Build a NVA project datamodel from a Cristin project datamodel.
      *
-     * @param cristinProject The model to convert from
      * @return a NvaProject converted from a CristinProject
      */
-    public static NvaProject mapCristinProjectToNvaProject(CristinProject cristinProject) {
-        NvaProject nvaProject = new NvaProject();
-
+    public NvaProject build() {
         // TODO: NP-2384: Remember to use setContext when serializing only a single NvaProject
         nvaProject.setId(buildUri(Constants.BASE_URL, cristinProject.cristinProjectId));
         nvaProject.setType(PROJECT_TYPE);
-        nvaProject.setIdentifier(
-            Collections.singletonList(Map.of(TYPE, CRISTIN_IDENTIFIER_TYPE, VALUE, cristinProject.cristinProjectId)));
-        if (cristinProject.title != null) {
-            nvaProject.setTitle(cristinProject.title.get(cristinProject.mainLanguage));
-            nvaProject.setAlternativeTitles(extractAlternativeTitles(cristinProject));
-        }
+        nvaProject.setIdentifier(createCristinIdentifier());
+        nvaProject.setTitle(extractMainTitle());
+        nvaProject.setAlternativeTitles(extractAlternativeTitles());
         nvaProject.setLanguage(buildUri(TEMPORARY_LANGUAGE_URL));
         nvaProject.setStartDate(cristinProject.startDate);
         nvaProject.setEndDate(cristinProject.endDate);
-        if (cristinProject.coordinatingInstitution != null) {
-            nvaProject.setCoordinatingInstitution(attempt(() ->
-                mapCristinInstitutionToNvaOrganization(cristinProject.coordinatingInstitution.institution))
-                .orElse(failure -> null));
+        nvaProject.setCoordinatingInstitution(extractCoordinatingInstitution());
+        nvaProject.setContributors(extractContributors());
+
+        return nvaProject;
+    }
+
+    private static NvaPerson mapCristinPersonToNvaPerson(Person person) {
+        if (person == null) {
+            return null;
         }
-        if (cristinProject.participants != null) {
-            nvaProject.setContributors(attempt(() ->
-                transformCristinPersonsToNvaContributors(cristinProject.participants))
-                .orElse(failure -> null));
+
+        NvaPerson identity = new NvaPerson();
```

Comment:
Should this be a new NvaPerson(person)?

then 

``` return nonNull(person) ? new NvaPerson(person) : null; ```

If we really want to return null, or Optional?

Reply:
Similarly for other methods